src/uagents/context.py

```diff
@@ -60,6 +61,7 @@ def __init__(
         self._resolver = resolve
         self._identity = identity
         self._queries = queries
+        self._session = session
```

Comment:
I think that 'ctx.session' should always return a UUID. So I would maybe generate a new one here is 'session is None'